src/Evolution/Conservative/ConservativeDuDt.hpp

```diff
@@ -0,0 +1,73 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \ingroup ConservativeGroup
+/// \brief Calculate \f$\partial u/\partial t\f$ for a conservative system.
+///
+/// The time evolution of the variables \f$u\f$ of any conservative
+/// system is given by \f$\partial_t u = - \partial_i F^i + S\f$,
+/// where \f$F^i\f$ are the fluxes as \f$S\f$ are the sources.
+///
+/// Source terms are only added for variables in the
+/// `System::sourced_variables` type list.
+template <typename System>
+struct ConservativeDuDt {
+ private:
+  template <typename TensorTagList, typename SourcedTagList>
+  struct apply_helper;
+
+  template <typename... TensorTags, typename... SourcedTags>
+  struct apply_helper<tmpl::list<TensorTags...>, tmpl::list<SourcedTags...>> {
+    static void function(
+        const gsl::not_null<db::item_type<TensorTags>*>... dt_u,
+        const db::item_type<TensorTags>&... div_u,
+        const db::item_type<SourcedTags>&... sources) noexcept {
+      const auto negate = [](const auto a, const auto& b) noexcept {
+        for (size_t i = 0; i < a->size(); ++i) {
+          (*a)[i] = -b[i];
+        }
+        return nullptr;
+      };
+      expand_pack(negate(dt_u, div_u)...);
```

Comment:
Should there be a useful error message for when a differing amount of dt_u's and div_u's are passed to this function?

Reply:
You will get the usual compiler error for when you try to call a function with the wrong arguments.